src/main/java/springData/controller/OfficeController.java

```diff
@@ -16,13 +20,47 @@
     @Autowired
     private OfficesService officesService;
 
+    @GetMapping
+    public @ResponseBody
+    Set<Office> findOfficeByTargetBetween(@RequestParam(value = "min", required = false) Integer minTarget,
+                                          @RequestParam(value = "max", required = false) Integer maxTarget) {
+        LOG.info("findByTargetBetween start, min={}, max={}", minTarget, maxTarget);
+        if (Objects.isNull(maxTarget) || Objects.isNull(minTarget)) {
+            LOG.debug("findOfficeByTargetBetween use getAllOffices");
+            return officesService.getAllOffices();
+        }
+        Set<Office> result = officesService.findByTargetBetween(minTarget, maxTarget);
+        LOG.info("findOfficeByTargetBetween done.");
+        return result;
+    }
+
     @GetMapping("/{id}")
     public @ResponseBody
-    Office getOfficeById(@PathVariable("id") int id) {
+    Office getOfficeById(@PathVariable("id") Integer id) {
         LOG.info("findOfficeById start, id={}", id);
-        Office result = officesService.findOfficeById(java.math.BigDecimal.valueOf(id));
+        Office result = officesService.findOfficeById(BigDecimal.valueOf(id));
         LOG.info("findOfficeById end");
-        throw new DeleteException("ERROR");
-//        return result;
+        return result;
+    }
+
+    @PutMapping("/{id}")
+    public void updateOfficeById(@PathVariable("id") Integer id, @RequestParam("city") String city) {
```

Comment:
Thats no good, that you can only update one field of entity. Maybe you should try to use json request?